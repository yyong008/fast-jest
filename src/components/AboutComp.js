import React from 'react'


const About = ({ name }) => {
	if (name) {
		return <div> About: {name}</div>
	}
	return <div>About: Component</div>
}

export default About;
