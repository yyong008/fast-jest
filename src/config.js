const config = {
	port: 8000,
	//... other
}

export default config;
