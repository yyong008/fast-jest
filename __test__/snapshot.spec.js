import React from "react";
import renderer from "react-test-renderer";
import AboutComp from "../src/components/AboutComp";

import config from "../src/config";

test("Snapshot Test config Object", () => {
  expect(config).toMatchSnapshot();
});

test("Snapshot Inline Test config Object", () => {
  expect(config).toMatchInlineSnapshot(`
    Object {
      "port": 8000,
    }
  `);
});

test("use react-test-renderer Component test", () => {
  const tree = renderer.create(<AboutComp />).toJSON();

  const treeName = renderer.create(<AboutComp name="react" />).toJSON();

  expect(tree).toMatchSnapshot();
  expect(treeName).toMatchSnapshot();
});

test("use react-test-renderer Component test Match Inline", () => {
  const tree = renderer.create(<AboutComp />).toJSON();

  const treeName = renderer.create(<AboutComp name="react" />).toJSON();

  expect(tree).toMatchInlineSnapshot(`
    <div>
      About: Component
    </div>
  `);
  expect(treeName).toMatchInlineSnapshot(`
    <div>
       About: 
      react
    </div>
  `);
});
