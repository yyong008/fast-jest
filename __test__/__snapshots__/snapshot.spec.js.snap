// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test config Object 1`] = `
Object {
  "port": 8000,
}
`;

exports[`use react-test-renderer Component test 1`] = `
<div>
  About: Component
</div>
`;

exports[`use react-test-renderer Component test 2`] = `
<div>
   About: 
  react
</div>
`;
